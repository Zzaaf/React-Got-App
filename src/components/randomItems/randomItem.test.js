import React from 'react';
import RandomItem from './randomItem';
import ErrorMessage from '../errorMessage';
import {shallow} from 'enzyme';

describe('Testing <RandomItem/>', () => {
    const char = shallow(<RandomItem/>);
    describe('Testing snap & state', () => {
        it('RandomItem has rendered correctly', () => {
            expect(char).toMatchSnapshot();
        });
        it('RandomItem state "char" is emtpy object', () => {
            expect(char.state().char).toBeObject();
        });
        it('RandomItem state "loading" is true', () => {
            expect(char.state().loading).toBeTruthy();
        });
        it('RandomItem state "error" is false', () => {
            expect(char.state().error).toBeFalsy();
        });
    });
    describe('Handlers tests', () => {
        it('Testing onCharLoaded', () => {
            char.instance().onCharLoaded();
            expect(char.state().loading).toBeFalsy();
        });
        it('Testing updateItem', () => {
            char.instance().updateItem();
            expect(char.state().loading).toBeFalsy();
        });
        it('Testing onError', () => {
            char.instance().onError(<ErrorMessage typeError={'404'}/>);
            expect(char.state().loading).toBeFalsy();
            expect(char.state().error).toBeTruthy();
        });
    });
});
