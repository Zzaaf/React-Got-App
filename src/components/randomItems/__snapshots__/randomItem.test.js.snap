// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <RandomItem/> Testing snap & state RandomItem has rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RandomItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Col
          lg={
            Object {
              "offset": 0,
              "size": 6,
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <ForwardRef
            className="rounded"
          >
            <Spinner />
          </ForwardRef>
        </Col>,
        <Col
          lg={
            Object {
              "size": 5,
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Button
            color="secondary"
            onClick={[Function]}
            outline={true}
            style={
              Object {
                "marginBottom": "300px",
              }
            }
            tag="button"
          >
            Remove character
          </Button>
        </Col>,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            className="rounded"
          >
            <Spinner />
          </ForwardRef>,
          "lg": Object {
            "offset": 0,
            "size": 6,
          },
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              null,
              null,
              <Spinner />,
            ],
            "className": "rounded",
          },
          "ref": null,
          "rendered": Array [
            null,
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    padding: 25px 25px 15px 25px;
    margin-bottom: 40px;
    background-color: #fff;
    h4 {
        margin-bottom: 20px;
        text-align: center;
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Button
            color="secondary"
            onClick={[Function]}
            outline={true}
            style={
              Object {
                "marginBottom": "300px",
              }
            }
            tag="button"
          >
            Remove character
          </Button>,
          "lg": Object {
            "size": 5,
          },
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Remove character",
            "color": "secondary",
            "onClick": [Function],
            "outline": true,
            "style": Object {
              "marginBottom": "300px",
            },
            "tag": "button",
          },
          "ref": null,
          "rendered": "Remove character",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Col
            lg={
              Object {
                "offset": 0,
                "size": 6,
              }
            }
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <ForwardRef
              className="rounded"
            >
              <Spinner />
            </ForwardRef>
          </Col>,
          <Col
            lg={
              Object {
                "size": 5,
              }
            }
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              color="secondary"
              onClick={[Function]}
              outline={true}
              style={
                Object {
                  "marginBottom": "300px",
                }
              }
              tag="button"
            >
              Remove character
            </Button>
          </Col>,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              className="rounded"
            >
              <Spinner />
            </ForwardRef>,
            "lg": Object {
              "offset": 0,
              "size": 6,
            },
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                null,
                null,
                <Spinner />,
              ],
              "className": "rounded",
            },
            "ref": null,
            "rendered": Array [
              null,
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "rules": Array [
                  "
    padding: 25px 25px 15px 25px;
    margin-bottom: 40px;
    background-color: #fff;
    h4 {
        margin-bottom: 20px;
        text-align: center;
    }
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Button
              color="secondary"
              onClick={[Function]}
              outline={true}
              style={
                Object {
                  "marginBottom": "300px",
                }
              }
              tag="button"
            >
              Remove character
            </Button>,
            "lg": Object {
              "size": 5,
            },
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Remove character",
              "color": "secondary",
              "onClick": [Function],
              "outline": true,
              "style": Object {
                "marginBottom": "300px",
              },
              "tag": "button",
            },
            "ref": null,
            "rendered": "Remove character",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
